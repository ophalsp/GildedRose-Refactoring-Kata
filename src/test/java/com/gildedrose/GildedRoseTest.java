package com.gildedrose;

import org.junit.jupiter.api.Test;

import static com.gildedrose.GildedRose.AGED_BRIE;
import static com.gildedrose.GildedRose.BACKSTAGE_PASSES_TO_A_TAFKAL_80_ETC_CONCERT;
import static com.gildedrose.GildedRose.CONJURED_MANA_CAKE;
import static com.gildedrose.GildedRose.SULFURAS_HAND_OF_RAGNAROS;
import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;
import static org.assertj.core.api.Assertions.assertThat;

public class GildedRoseTest {

    @Test
    void testLegacyCode() {
        verifyAllCombinations(this::update,
                new String[] {"foo", AGED_BRIE, SULFURAS_HAND_OF_RAGNAROS, BACKSTAGE_PASSES_TO_A_TAFKAL_80_ETC_CONCERT},
                new Integer[] {0, 5, 6, 10, 11}, //Use this sellIn values, defined with the help of code coverage
                new Integer[] {0, 1, 2, 49, 50}); //Use this quality values, defined with the help of code coverage
    }

    @Test
    void testConjured() {
        Item conjuredManaCake = update(CONJURED_MANA_CAKE, 4, 6);
        assertConjuredItem(conjuredManaCake, 3, 4);
        conjuredManaCake = update(conjuredManaCake);
        assertConjuredItem(conjuredManaCake, 2, 2);
        conjuredManaCake = update(conjuredManaCake);
        assertConjuredItem(conjuredManaCake, 1, 0);
        conjuredManaCake = update(conjuredManaCake);
        assertConjuredItem(conjuredManaCake, 0, 0);
        conjuredManaCake = update(conjuredManaCake);
        assertConjuredItem(conjuredManaCake, -1, 0);
    }

    private Item update(String name, int sellIn, int quality) {
        Item item = new Item(name, sellIn, quality);
        return update(item);
    }

    private Item update(Item item) {
        GildedRose app = new GildedRose(new Item[] {item});
        app.updateQuality();
        return item;
    }

    private void assertConjuredItem(Item item, int expectedSellIn, int expectedQuality) {
        assertThat(item.name).isEqualTo(CONJURED_MANA_CAKE);
        assertThat(item.sellIn).isEqualTo(expectedSellIn);
        assertThat(item.quality).isEqualTo(expectedQuality);
    }
}
