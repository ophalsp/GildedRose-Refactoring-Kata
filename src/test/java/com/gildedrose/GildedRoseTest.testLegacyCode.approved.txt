[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 2] => foo, -1, 0 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 50] => foo, -1, 48 
[foo, 5, 0] => foo, 4, 0 
[foo, 5, 1] => foo, 4, 0 
[foo, 5, 2] => foo, 4, 1 
[foo, 5, 49] => foo, 4, 48 
[foo, 5, 50] => foo, 4, 49 
[foo, 6, 0] => foo, 5, 0 
[foo, 6, 1] => foo, 5, 0 
[foo, 6, 2] => foo, 5, 1 
[foo, 6, 49] => foo, 5, 48 
[foo, 6, 50] => foo, 5, 49 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 1] => foo, 9, 0 
[foo, 10, 2] => foo, 9, 1 
[foo, 10, 49] => foo, 9, 48 
[foo, 10, 50] => foo, 9, 49 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 1] => foo, 10, 0 
[foo, 11, 2] => foo, 10, 1 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 50] => foo, 10, 49 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 2] => Aged Brie, -1, 4 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 2] => Aged Brie, 4, 3 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 6, 0] => Aged Brie, 5, 1 
[Aged Brie, 6, 1] => Aged Brie, 5, 2 
[Aged Brie, 6, 2] => Aged Brie, 5, 3 
[Aged Brie, 6, 49] => Aged Brie, 5, 50 
[Aged Brie, 6, 50] => Aged Brie, 5, 50 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 1] => Aged Brie, 9, 2 
[Aged Brie, 10, 2] => Aged Brie, 9, 3 
[Aged Brie, 10, 49] => Aged Brie, 9, 50 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 2] => Aged Brie, 10, 3 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 0, 2] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 80 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 80 
[Sulfuras, Hand of Ragnaros, 5, 1] => Sulfuras, Hand of Ragnaros, 5, 80 
[Sulfuras, Hand of Ragnaros, 5, 2] => Sulfuras, Hand of Ragnaros, 5, 80 
[Sulfuras, Hand of Ragnaros, 5, 49] => Sulfuras, Hand of Ragnaros, 5, 80 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 80 
[Sulfuras, Hand of Ragnaros, 6, 0] => Sulfuras, Hand of Ragnaros, 6, 80 
[Sulfuras, Hand of Ragnaros, 6, 1] => Sulfuras, Hand of Ragnaros, 6, 80 
[Sulfuras, Hand of Ragnaros, 6, 2] => Sulfuras, Hand of Ragnaros, 6, 80 
[Sulfuras, Hand of Ragnaros, 6, 49] => Sulfuras, Hand of Ragnaros, 6, 80 
[Sulfuras, Hand of Ragnaros, 6, 50] => Sulfuras, Hand of Ragnaros, 6, 80 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 80 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros, 10, 80 
[Sulfuras, Hand of Ragnaros, 10, 2] => Sulfuras, Hand of Ragnaros, 10, 80 
[Sulfuras, Hand of Ragnaros, 10, 49] => Sulfuras, Hand of Ragnaros, 10, 80 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 80 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 80 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 80 
[Sulfuras, Hand of Ragnaros, 11, 2] => Sulfuras, Hand of Ragnaros, 11, 80 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 80 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 80 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 2] => Backstage passes to a TAFKAL80ETC concert, 4, 5 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Backstage passes to a TAFKAL80ETC concert, 6, 2] => Backstage passes to a TAFKAL80ETC concert, 5, 4 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert, 9, 3 
[Backstage passes to a TAFKAL80ETC concert, 10, 2] => Backstage passes to a TAFKAL80ETC concert, 9, 4 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 2] => Backstage passes to a TAFKAL80ETC concert, 10, 3 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
